src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/Base64Encoder.cs

```diff
@@ -14,15 +16,22 @@ public byte[] Decode(byte[] payload)
             var buffer = new ReadOnlyMemory<byte>(payload);
             LengthPrefixedTextMessageParser.TryParseMessage(ref buffer, out var message);
 
-            return Convert.FromBase64String(Encoding.UTF8.GetString(message.ToArray()));
+            Span<byte> decoded = new byte[Base64.GetMaxDecodedFromUtf8Length(message.Length)];
+            var status = Base64.DecodeFromUtf8(message.Span, decoded, out _, out var written);
+            Debug.Assert(status == OperationStatus.Done);
+
+            return decoded.Slice(0, written).ToArray();
         }
 
         public byte[] Encode(byte[] payload)
         {
-            var buffer = Encoding.UTF8.GetBytes(Convert.ToBase64String(payload));
+            Span<byte> buffer = new byte[Base64.GetMaxEncodedToUtf8Length(payload.Length)];
+
+            var status = Base64.EncodeToUtf8(payload, buffer, out _, out var written);
+            Debug.Assert(status == OperationStatus.Done);
             using (var stream = new MemoryStream())
             {
-                LengthPrefixedTextMessageWriter.WriteMessage(buffer, stream);
+                LengthPrefixedTextMessageWriter.WriteMessage(buffer.Slice(0, written), stream);
```

Comment:
Probably not part of this PR, but we should look at removing the need for stream

Reply:
This actually should be part of this PR. Fixing.

Reply:
Decided to do it separately...